packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -1860,14 +1860,17 @@ operations will be generated by the CLI.`,
         // check if owner is enabled in auth
         const ownerRules = rules.filter(rule => rule.allow === OWNER_AUTH_STRATEGY);
         const needsDefaultOwnerField = ownerRules.find(rule => !rule.ownerField);
-        const hasStaticGroupAuth = rules.find(rule => rule.allow === GROUPS_AUTH_STRATEGY && !rule.groupsField);
+        // if there are any public, private, and/or static group rules then the owner argument is optional
```

Comment:
I don't really follow this comment with the code below. Can you expound a bit upon the problem you're solving, and the desired future state, in the pull request description?

Reply:
The owner argument becomes non-nullable if it's only auth rule on subscriptions, otherwise if there are other auth rules (such as public, private, static group rules) the owner argument becomes optional. This way if an argument such as `onCreatePost` is called with IAM auth the optional argument owner is not required because it was authorized via IAM.